File: languages/python/bloock/entity/key/access_control_totp.py
from __future__ import annotations

import bloock._bridge.proto.keys_entities_pb2 as proto


class AccessControlTotp:
    """
    Represents a Time-based One-Time Password ([TOTP](https://datatracker.ietf.org/doc/html/rfc6238)) code used for access control.
    """
    code = None

    def __init__(self, code) -> None:
        """
        Constructs an AccessControlTotp object with the specified parameters.
        :type code: object
        :rtype: object
        """
        self.code = code

    def to_proto(self) -> proto.AccessControlTotp:
        return proto.AccessControlTotp(
            code=self.code
        )

File: languages/php/Bloock/Entity/Key/AccessControlTotp.php
<?php

namespace Bloock\Entity\Key;

/**
 * Represents a Time-based One-Time Password ([TOTP](https://datatracker.ietf.org/doc/html/rfc6238)) code used for access control.
 */
class AccessControlTotp
{
    public string $code;

    /**
     * Constructs an AccessControlTotp object with the specified parameters.
     * @param string $code
     */
    public function __construct(string $code)
    {
        $this->code = $code;
    }

    public function toProto(): \Bloock\AccessControlTotp
    {
        $s = new \Bloock\AccessControlTotp();

        $s->setCode($this->code);

        return $s;
    }
}
